src/main/java/com/google/cloud/genomics/dataflow/functions/KeyReadsFn.java

```diff
@@ -52,15 +60,31 @@ public KeyReadsFn() {
   
   @Override
   public void startBundle(Context c) {
-  lociPerShard = c.getPipelineOptions()
+    lociPerShard = c.getPipelineOptions()
       .as(Options.class)
       .getLociPerWritingShard();
+    lastPos = -1;
+    count = 0;
+  }
+  
+  @Override
+  public void finishBundle(Context c) {
+    LOG.info("KeyReadsDone: Processed " + count + " reads" + "min=" + minPos + 
+        " max=" + maxPos);
   }
 
   @Override
   public void processElement(DoFn<Read, KV<Contig, Read>>.ProcessContext c)
     throws Exception {
     final Read read = c.element();
+    long pos = read.getAlignment().getPosition().getPosition();
+    if (lastPos > pos) {
+      LOG.info("Out of order read at pos "+ lastPos + " " + pos);
```

Comment:
It's fine for the read positions in a BAM to be unsorted.  In fact you can add a followup merge/reduce step in the dataflow to have the reads sorted by position if you want.